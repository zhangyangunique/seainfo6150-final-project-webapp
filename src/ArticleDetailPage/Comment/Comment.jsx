import React, { useState } from "react";
import styles from "./Comment.module.css";

const Comment = () => {
	const [submittedForm, setSubmittedForm] = useState();
	function onSubmit(e) {
		e.preventDefault();
		const data = new FormData(e.target);
		setSubmittedForm(data);
	};

	return (
		<div className={styles.comment}>
			<h2>Feel free to leave a comment</h2>
			{
				submittedForm ? (
					<div className={styles.commentText}>You have successfully left the comment: {submittedForm.get("commentText")}</div>
				) : (
						<form className={styles.commentForm} onSubmit={onSubmit}>
							<textarea className={styles.textarea} name="commentText"></textarea>
							<input className={styles.submitButton} type="submit" />
						</form>
					)
			}
		</div>
	)
}

export default Comment;