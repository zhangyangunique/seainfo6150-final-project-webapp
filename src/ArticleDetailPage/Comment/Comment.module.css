.comment {
	font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
	margin-top: 100px;
}

.commentForm {
	display: block;
	width: 70%;
	margin: 0 auto;
}

.textarea {
	width: 100%;
	height: 200px;
	margin-bottom: 20px;
}

.submitButton {
	width: 100%;
	height: 40px;
	background-color: black;
	color: white;
	border: none;
	border-radius: 2px;
}

.commentText {
	text-align: center;
	font-size: 30px;
}