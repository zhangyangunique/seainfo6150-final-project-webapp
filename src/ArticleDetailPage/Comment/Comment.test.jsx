import React from "react";
import { render } from "@testing-library/react";
import Comment from "./Comment";

describe("Comment tests", () => {
    it("renders correctly", () => {
        const { container } = render(<Comment />);
        expect(container).toMatchSnapshot();
    });
});