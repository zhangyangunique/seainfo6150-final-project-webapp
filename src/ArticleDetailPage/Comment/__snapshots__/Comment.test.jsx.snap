// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment tests renders correctly 1`] = `
<div>
  <div
    class="comment"
  >
    <h2>
      Feel free to leave a comment
    </h2>
    <form
      class="commentForm"
    >
      <textarea
        class="textarea"
        name="commentText"
      />
      <input
        class="submitButton"
        type="submit"
      />
    </form>
  </div>
</div>
`;
